pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1357,7 +1358,14 @@ protected void handleProducer(final CommandProducer cmdProducer) {
                     });
 
                     schemaVersionFuture.thenAccept(schemaVersion -> {
-                        topic.checkIfTransactionBufferRecoverCompletely(isTxnEnabled).thenAccept(future -> {
+                        CompletableFuture<Void> checkTxnRecoverFuture;
+                        if (isTxnEnabled) {
+                            checkTxnRecoverFuture = topic.checkIfTransactionBufferRecoverCompletely();
+                        } else {
+                            checkTxnRecoverFuture = TransactionBuffer.TRANSACTION_NOT_ENABLED;
+                        }
```

Comment:
It's better to check `isTxnEnable` in `checkIfTransactionBufferRecoverCompletely `, it will make the logic in `ServerCnx` more clear